src/NBitcoin/UInt256.cs

```diff
@@ -435,11 +506,23 @@ public uint GetLow32()
 
         public override int GetHashCode()
         {
-            return (int)this.pn0;
+            int hash = 17;
```

Comment:
There was actually a good reason why we just return pn0, for perf its significantly better and pn0 should in theory be as random as hashing all the rest of the value

Reply:
I've reverted back to pn0 as per the original code